opm/simulators/wells/StandardWell_impl.hpp

```diff
@@ -1872,7 +1874,16 @@ namespace Opm
         const auto& summaryState = ebosSimulator.vanguard().summaryState();
         if (!Base::wellHasTHPConstraints(summaryState) || bhp_controlled_well) {
             // get the bhp value based on the bhp constraints
-            const double bhp = this->mostStrictBhpFromBhpLimits(summaryState);
+            double bhp = this->mostStrictBhpFromBhpLimits(summaryState);
+
+            // in some very special cases the bhp pressure target are
```

Comment:
Can you explain more about under what special cases, the BHP constraint can be violated? I have been always thinking  BHP constraint is a pretty hard constraint. 

And also, have you seen the situations that negative well potentials are resulted?

Reply:
We compute the well potentials before we check the constrains and also we don't check the constrains in our getWellConverence method. Negative potentials can happen for new wells for instance. Say you open a injector that is supposed to give pressure support to your producer. The solution will need some newton iterations before the pressure increase due to injection is propagated to the producer.

Reply:
Then there is a problem here. With the code, you are computing the well potentials based on the old bhp value fromthe well state instead of the BHP constraint. Will it be against the definition of the well potentials? or I misunderstood it?

Reply:
Yes it will be against the definition, but it is only a temporary solution. When the simulator starts to iterate the bhp limit will be violated and ws.bhp = bhp and we are on track again. The point of this temporary solution is to be able to keep the well open even if it initially (i.e. before the newton loop) thinks it does not have potential to keep open it may get the potential it need to produce when the injector has done its job of increasing the pressure. I have tested this on several model and the results seems to be ok, also for the potentials. Even a slightly wrong solution for the potentials is better than shutting the well completely if it is supposed to stay open.

Reply:
I am a little too slow on this and need a description of the scenario this fix is targeting. Maybe we will be able to discuss it in person tomorrow.